libraries/WiFiClientSecure/src/WiFiClientSecure.h

```diff
@@ -62,6 +62,7 @@ class WiFiClientSecure : public WiFiClient
     bool loadCertificate(Stream& stream, size_t size);
     bool loadPrivateKey(Stream& stream, size_t size);
     bool verify(const char* fingerprint, const char* domain_name);
+    void setHandshakeTimeout(unsigned long handshake_timeout);
```

Comment:
It probably should be noted that this is expected to be seconds and not milliseconds.

Reply:
yeah I converted them to millis in the function.

Reply:
most likely some sort of note here will help others not to pass the wrong type. Maybe even in the method itself do some bounds checking to ensure it doesn't wait forever (ie: cap it at 3600 seconds)

Reply:
@atanisoft  default timeout will be 2 min. It is user wish if he wants to change it.